routes/products.js

```diff
@@ -0,0 +1,203 @@
+const knex = require("../database/index");
+
+let error = "";
+let success = "";
+
+router.get("/search", (req, res) => {
+  res.render("products/search", { error: error });
+  error = "";
+});
+
+router.get("/delete", (req, res) => {
+  res.render("products/delete", { error: error, success: success });
+  error = "";
+  success = "";
+});
+
+router.get("/new", (req, res) => {
+  res.render("products/add", { error: error });
+  error = "";
+});
+
+router.get("/edit", (req, res) => {
+  knex
+    .select("*")
+    .from("products")
+    .orderBy("id", "asc")
+    .then(results => {
+      res.render("products/edit", {
+        products: results,
+        error: error
+      });
+      error = "";
+    });
+});
+
+router.put("/edit", (req, res) => {
+  let itemID = parseInt(req.body.id);
+  let itemName = req.body.name;
+  let itemPrice = parseInt(req.body.price);
+  let itemInventory = parseInt(req.body.inventory);
+
+  knex("products")
+    .where("id", itemID)
+    .update({
+      name: itemName,
+      price: itemPrice,
+      inventory: itemInventory
+    })
+    .returning("*")
+    .then(results => {
+      if (
+        req.body.name === "" ||
+        req.body.price === "" ||
+        req.body.inventory === ""
+      ) {
+        throw err;
+      } else {
+        res.render("products/product", { products: results });
+      }
+    })
+    .catch(err => {
+      error = "Please fill in all fields";
+      res.redirect("/products/edit");
+    });
+});
+
+router.delete("/delete", (req, res) => {
+  let itemID = parseInt(req.body.id);
+
+  knex("products")
+    .where("id", itemID)
+    .del()
+    .then(results => {
+      if (itemID === "" || results == 0) {
+        throw err;
+      } else {
+        success = "Successfully Deleted!";
+        res.redirect("/products/delete");
+      }
+    })
+    .catch(err => {
+      error = "Could not find your product. Try again.";
+      res.redirect("/products/delete");
+    });
+});
+
+router.delete("/delete/:id", (req, res) => {
+  let itemID = parseInt(req.params.id);
+
+  knex("products")
+    .where("id", itemID)
+    .del()
+    .then(results => {
+      if (itemID === "" || results == 0) {
+        throw err;
+      } else {
+        success = "Successfully Deleted!";
+        res.redirect("/products/delete");
+      }
+    })
+    .catch(err => {
+      error = "Could not find your product. Try again.";
+      res.redirect("/products/delete");
+    });
+});
+
+router.get("/delete/:id", (req, res) => {
+  let itemID = parseInt(req.params.id);
+
+  knex("products")
+    .where("id", itemID)
+    .del()
+    .then(results => {
+      if (itemID === "" || results == 0) {
+        throw err;
+      } else {
+        success = "Successfully Deleted!";
+        res.redirect("/products/delete");
+      }
+    })
+    .catch(err => {
+      error = "Could not find your product. Try again.";
+      res.redirect("/products/delete");
+    });
+});
+
+router.get("/fetch", (req, res) => {
```

Comment:
Not sure what this route is for. Getting a item should only be on the GET `/:id` route